### ISSUE TITLE
Webview gets stuck to cursor on draggable area

### ISSUE THREAD

**Issue by der-lukas** [skpm/sketch-module-web-view#100]
When the webview isn't focused and I want to drag it without focusing the window before, it often gets "stuck" to the cursor-position and I can't get is loose without killing the webview!

![webview_draggable-area-bug 2019-05-06 17_38_40](https://user-images.githubusercontent.com/5270044/57236626-d3248e80-7025-11e9-953d-75735541bb02.gif)

Here's the draggable area HTML-markup (first element inside the body)

```
<div class="row header" data-app-region="drag">
```

and here's the window config:

```
const options = {
        identifier: 'smartStyleWindow',
        width: 290,
        height: 400,
        frame: false,
        alwaysOnTop: true,
        hasShadow: true,
        acceptsFirstMouse: true,
        show: false,
        resizable: false,
        closable: false
    }
```

... so nothing special there...

Any idea what's going on here?

**Comment by mathieudutour** [comment]
I'm not able to reproduce but I'm not too surprised tbh.

Both the `acceptsFirstMouse` and `data-app-region` are pretty big hacks when I inject some javascript in the document:
- `acceptsFirstMouse` dispatch a `click` event on focus
- `data-app-region` listen to `mousedown` and `mouseup` to send events to the plugin which moves the window

It might be possible that the two conflict with each other somehow 🤔

**Comment by rodomeista** [comment]
+1 I've seen this happen before as well

**Comment by mathieudutour** [comment]
Could you check if v3.0.7 fixes this?

**Comment by xsfour** [comment]
I tried in v3.0.7 and got a `mouseup` event, which will stop moving the window, on the element

**Comment by xsfour** [comment]
In my opinion, we need take care of the old `animaitonId` here:
https://github.com/skpm/sketch-module-web-view/blob/800665c0f271a0fb345e664d26c84f8d9b3097fb/lib/movable-area.js#L21

Otherwise, if we lost a `mouseup` event and get a `mousedown` event again, the old animation request would exist infinitely. Sorry, I can't explain it clearly because of my poor English.

Or we can listen to `mousemove` event instead of using `requestAnimationFrame`. Here is my solution:
```javascript
(function () {
    document.addEventListener('mousedown', onMouseDown);
    function shouldDrag(target) {
      if (!target || (target.dataset || {}).appRegion === "no-drag") {
        return false
      } if ((target.dataset || {}).appRegion === "drag") {
        return true
      }
      return shouldDrag(target.parentElement)
    };
    function onMouseDown(e) {
      if (e.button !== 0 || !shouldDrag(e.target)) {
        return
      }
      window.postMessage('${CONSTANTS.START_MOVING_WINDOW}');
      document.addEventListener('mouseup', onMouseUp);
      document.addEventListener('mousemove', onMouseMove);
    };
    function onMouseUp(e) {
      document.removeEventListener('mousemove', onMouseMove);
      document.removeEventListener('mouseup', onMouseUp);
      window.postMessage('${CONSTANTS.STOP_MOVING_WINDOW}');
    };
    function onMouseMove(e) {
      window.postMessage('${CONSTANTS.MOVE_WINDOW}');
    }
  })()
```

**Comment by xsfour** [comment]
But both solutions will met the same problem: we'll lose `mouseup` event if we release the mouse button outside the webview window.

I found the problem only occurs in webview but not Safari or Chrome window.

**Comment by mathieudutour** [comment]
Ah!

> mouseup will only fire when performed within the browser, which can lead to losing track of the button state.

Looks like we can check during the `mousemove` event if the left button is pressed:

```js
const leftButtonPressed = e.buttons === undefined 
    ? e.which === 1 // safari only
    : e.buttons === 1 // standard
```

If `leftButtonPressed` is false, then we should consider that we missed the `mouseup` event.

So what we can do is, listen to `mousemove` (debounced inside a `requestAnimationFrame` to avoid killing performance) and check for `leftButtonPressed`

Does someone want to take a stab at it?

**Comment by xsfour** [comment]
I found the hidden title bar may be the main cause of the problem because it may covers the draggable region.
I try to fix this by #105
